packages/server/lib/util/settings.js

```diff
@@ -144,6 +148,16 @@ module.exports = {
     .catch({ code: 'ENOENT' }, () => {
       return this._write(file, {})
     }).then((json = {}) => {
+      if (this.isComponentTesting(options) && 'component' in json) {
```

Comment:
Is there any validation for the `component` and `e2e` properties? What happens if someone passes `component: true` or `component: 'foo'`? I'd like to see some tests covering that too.

Reply:
I have now added validation in the same way we validate all the other options. We cannot validate here if we want to show the errors to the user in the Desktop GUI app. The validation happens later on. 

With this change, errors are correctly shown:

![image](https://user-images.githubusercontent.com/19196536/111564821-1e915180-87e6-11eb-85c0-910921c8089f.png)

Reply:
It shouldn't be too much work to fully validate each object. Might just need to move [the validate function](https://github.com/cypress-io/cypress/blob/8f7b596c8691c6442a7bcdf3a1976a6d78f05b3a/packages/server/lib/config.js#L85) into [util/validation.js](https://github.com/cypress-io/cypress/blob/8f7b596c8691c6442a7bcdf3a1976a6d78f05b3a/packages/server/lib/util/validation.js) so it can referenced in place of `v.isPlainObject` in lib/config_options.js. Other than that, maybe enhance it so the error message makes it clear it's referencing the nested property, like:

> Expected `component.port` to be a number. Instead the value was: `"1234"`

Reply:
I think this should block. It should be a consistent experience whether the property is top-level or nested.